### ISSUE TITLE
pzdeaths command error

### ISSUE THREAD

**Issue by AWL-Gaming** [rfalias/project_zomboid_bot#11]
The !pzdeaths command is not working. Geting an "Invalid Form Body" error 

```
Ignoring exception in command pzdeaths:
Traceback (most recent call last):
  File "C:\Python310\lib\site-packages\discord\ext\commands\core.py", line 85, in wrapped
    ret = await coro(*args, **kwargs)
  File "Z:\Bots\project_zomboid_bot\pzbot.py", line 453, in pzdeaths
    await ctx.send(results)
  File "C:\Python310\lib\site-packages\discord\abc.py", line 1065, in send
    data = await state.http.send_message(channel.id, content, tts=tts, embed=embed,
  File "C:\Python310\lib\site-packages\discord\http.py", line 254, in request
    raise HTTPException(r, data)
discord.errors.HTTPException: 400 Bad Request (error code: 50035): Invalid Form Body
In content: Must be 4000 or fewer in length.

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "C:\Python310\lib\site-packages\discord\ext\commands\bot.py", line 939, in invoke
    await ctx.command.invoke(ctx)
  File "C:\Python310\lib\site-packages\discord\ext\commands\core.py", line 863, in invoke
    await injected(*ctx.args, **ctx.kwargs)
  File "C:\Python310\lib\site-packages\discord\ext\commands\core.py", line 94, in wrapped
    raise CommandInvokeError(exc) from exc
discord.ext.commands.errors.CommandInvokeError: Command raised an exception: HTTPException: 400 Bad Request (error code: 50035): Invalid Form Body
In content: Must be 4000 or fewer in length.
```

**Comment by rfalias** [comment]
Check your LOGS_PATH in .env
It's discord complaining that the text is too long to send over discord. It seems what ever it's finding is too long. 
I added a quick print to that command so you can check the python logs and see what it's trying to return

**Comment by AWL-Gaming** [comment]
> Check your LOGS_PATH in .env It's discord complaining that the text is too long to send over discord. It seems what ever it's finding is too long. I added a quick print to that command so you can check the python logs and see what it's trying to return

Logs are set to LOG_PATH="C:\Users\Administrator\Zomboid\Logs" in the .env file
I can see everything logged including the individual player logs, join and leaves. Tried the new commit and it's now printing all deaths correctly inside the Python console window. Current text is at 10052 characters which is way beyond the 4000 limit of Discord.

**Comment by rfalias** [comment]
Is the text actually correct on the death report?  Can you paste it here?
I guess if you have a really busy server it could be possible. If so I can paginate it and send multiple discord messages if that would be better.

**Comment by AWL-Gaming** [comment]
> Is the text actually correct on the death report? Can you paste it here? I guess if you have a really busy server it could be possible. If so I can paginate it and send multiple discord messages if that would be better.

Yes it is correct. We have a somewhat busy server with a lot of daily players on.
Here's the full text:

```
Jubei has died 11 times
Draggon has died 5 times
Circles has died 1 times
Teo has died 1 times
admin has died 4 times
Zeke33 has died 17 times
Armery has died 4 times
WalkingBuffet has died 9 times
Petyyz has died 3 times
Dragg0n has died 3 times
Typhoon has died 1 times
adlk has died 6 times
Nine has died 9 times
seastar has died 2 times
Rawlings has died 5 times
dab has died 3 times
Mac has died 1 times
Ashi has died 13 times
Mirrenis has died 9 times
Ranger has died 23 times
124 has died 6 times
Frenno has died 4 times
brt has died 1 times
Salmon has died 3 times
Ruth has died 1 times
bandar has died 9 times
Profile has died 5 times
Jiorgon has died 2 times
Tootaroot has died 3 times
Mini has died 2 times
whitefox has died 2 times
ManyLegs has died 3 times
kubfrid has died 8 times
karaboga has died 6 times
Ethalider has died 2 times
NigelT has died 1 times
Machine has died 3 times
High has died 16 times
Breaky has died 3 times
Drdoctor has died 9 times
Violet has died 4 times
BerryFPS has died 5 times
EddyThe has died 2 times
lulu has died 5 times
Big has died 3 times
Devilzblade has died 1 times
SadMahiro has died 2 times
patthehetic has died 1 times
Alzworth has died 1 times
vib has died 2 times
woof has died 5 times
Pickled has died 29 times
pavan has died 34 times
maxee has died 9 times
Milchdieb has died 1 times
MrBrown has died 8 times
Sillath has died 1 times
Shee has died 2 times
FedOTO has died 7 times
Beniamin has died 1 times
Rath has died 3 times
Krul has died 1 times
Max has died 14 times
Abyss has died 1 times
Drizzyyy has died 8 times
rvia has died 2 times
Maltese has died 10 times
Gibbo has died 6 times
pulse2 has died 1 times
Magma has died 1 times
Xiod has died 13 times
Wolfee has died 2 times
Grey has died 8 times
datafara has died 3 times
Rqez has died 6 times
DudeDissmus has died 3 times
Snuky has died 5 times
Podliva has died 7 times
cclarkemkx has died 16 times
chadboi has died 2 times
Mikel has died 1 times
Walter has died 7 times
connor has died 9 times
Bongo has died 5 times
MainProtagonist has died 2 times
Grotebeer has died 1 times
Konratou has died 1 times
Sterh has died 3 times
Defiant has died 3 times
asdfasdfasdf has died 2 times
brede has died 5 times
Kilda1 has died 1 times
Cerventus has died 1 times
steve has died 1 times
Xilia has died 1 times
Sonox has died 2 times
kinghew has died 4 times
Z9Ryan has died 1 times
wheelchair has died 1 times
Hepok has died 1 times
om has died 1 times
bonkers has died 1 times
Ottar has died 7 times
Baccano has died 12 times
Leshiy has died 2 times
nailoo45 has died 1 times
John has died 1 times
Necrilimoglsed has died 7 times
schattenov has died 3 times
wizzah has died 2 times
Jingleheimer has died 1 times
gary has died 1 times
yewhan has died 2 times
MiLo has died 2 times
LowroX has died 1 times
Mabor85 has died 3 times
Roadrunner has died 1 times
Dyrt has died 1 times
Dubaizz has died 11 times
Kannoli has died 5 times
uncut has died 4 times
KILLSHOT has died 7 times
Tobimix has died 1 times
Robercik has died 2 times
ReBo has died 2 times
Hunter_S has died 1 times
Mieko has died 9 times
Monty has died 4 times
qwertz has died 2 times
Rik has died 3 times
Yukio has died 11 times
Van has died 1 times
nyx has died 2 times
Kbob has died 5 times
morpheuz has died 6 times
Marnov has died 1 times
DVT00 has died 1 times
Szeligus has died 3 times
vaster has died 2 times
VacPop has died 1 times
Destrox has died 2 times
Oozyxxxhybrid has died 1 times
Alex has died 1 times
Vipe has died 1 times
LokomotivHAFT has died 1 times
bbbbbb has died 1 times
Aurora has died 1 times
OrBye has died 8 times
LivizCZ has died 4 times
Toffel has died 1 times
Norman has died 2 times
Rumor has died 1 times
Chocoyote has died 3 times
MaxxVonSchutle has died 2 times
Carl has died 2 times
frankiesome has died 1 times
Kinns has died 4 times
DometrixTv has died 10 times
bananaonana has died 1 times
Szonik has died 8 times
datingtibbers has died 3 times
Julie has died 7 times
DGX has died 8 times
PabloEscoBAR has died 8 times
Ogreballs has died 3 times
Morfij has died 1 times
Duskus has died 2 times
warp has died 12 times
Negan has died 6 times
Bane has died 1 times
DanotheDino has died 5 times
Gasenjoyer has died 1 times
aponus has died 6 times
Agent has died 3 times
flugi has died 2 times
NotchBoii has died 2 times
Pumukiff has died 1 times
Cosmictrip has died 4 times
SmauglysLT has died 9 times
Janis has died 1 times
joker has died 1 times
Fratii has died 7 times
vek has died 1 times
Glower has died 1 times
Smyrek has died 4 times
Shadain has died 9 times
DerTibbels has died 4 times
WIKID has died 3 times
Chrixzi has died 1 times
luxus1988 has died 3 times
SwallowMeDaddy has died 1 times
VvojteroxX has died 3 times
bink has died 1 times
Shaybos has died 3 times
Rasmus has died 1 times
Joe has died 4 times
TeD has died 1 times
Mel has died 1 times
Matty has died 2 times
Navi has died 3 times
GrubyKuba has died 2 times
ducik has died 1 times
Boneflower has died 6 times
Cabal2468 has died 4 times
BARNY has died 6 times
FrostedFoxes has died 5 times
Tobman has died 8 times
ChupacabraKenobi has died 3 times
Ducker has died 2 times
murtus has died 1 times
Adam has died 1 times
Bree has died 1 times
Equinox has died 3 times
RAPTOR has died 1 times
oral has died 1 times
andrea has died 1 times
Smiley has died 3 times
Adr1nz has died 1 times
Gerald has died 6 times
deno has died 5 times
HeQu has died 3 times
lilkusia has died 1 times
Shekel has died 2 times
Flirtze has died 3 times
Sgt has died 1 times
Salty has died 2 times
Kaz has died 7 times
Foxsnow has died 1 times
Cookie has died 1 times
yahoo has died 1 times
Oshiolga36 has died 3 times
uky has died 1 times
murat has died 1 times
Bidaze has died 10 times
Average has died 3 times
Boact has died 1 times
SleepingScout has died 4 times
Myor has died 1 times
unig has died 1 times
Frank has died 6 times
Newt has died 1 times
Swaggy has died 2 times
Pavel has died 9 times
Wootz has died 17 times
WatchMaster has died 2 times
BowserB has died 3 times
Zureyne has died 2 times
Alf has died 1 times
akraprovik has died 4 times
Postal has died 1 times
CrazyAshez has died 1 times
Sleepy has died 1 times
Exitazz1337 has died 2 times
TheRedReaper has died 1 times
Johnny has died 1 times
Dark has died 2 times
Frankie has died 1 times
oni has died 2 times
creep5021 has died 1 times
futi has died 4 times
DarkSpy has died 1 times
Strach has died 6 times
Ben has died 6 times
xedir has died 2 times
Man-Man has died 3 times
Adrian has died 1 times
Rick has died 2 times
Popsyheart has died 1 times
Essah has died 2 times
kalasH has died 4 times
DoktorTed has died 3 times
kami has died 17 times
NJS has died 4 times
zootnskoot has died 2 times
Ruban has died 2 times
LambchopGaming has died 3 times
Kyu has died 5 times
CPTWoodhouse has died 1 times
Lynx has died 7 times
Kyy has died 12 times
Worlds has died 12 times
Gage has died 2 times
Moist has died 1 times
krassimir111 has died 4 times
Dilli has died 7 times
Sekler has died 6 times
Wing has died 1 times
WolfpackXx has died 1 times
Frontier has died 1 times
CoinLords has died 6 times
Sokratis has died 2 times
obez has died 1 times
PsychoDave has died 1 times
Pope has died 1 times
Elans has died 1 times
kebabcizade2 has died 2 times
Sempai has died 4 times
PlumedArcher has died 2 times
Bozo018 has died 13 times
frostedc130 has died 6 times
BigBadPanda has died 3 times
phatd has died 1 times
izahad has died 2 times
main has died 4 times
magmaheady has died 1 times
Zed has died 4 times
Plato(n) has died 11 times
TrelogriaGR has died 1 times
kafe has died 1 times
play0r has died 1 times
Dave has died 3 times
buckingb has died 1 times
Lavush has died 3 times
Teragos has died 1 times
KingFerra has died 1 times
xDave has died 1 times
QQhammer has died 1 times
kutt has died 8 times
moonlight_frag has died 2 times
Harlequin has died 1 times
Meru has died 1 times
chosaw has died 1 times
Blatantl has died 2 times
DelForelle has died 1 times
Sly_4 has died 2 times
Jay has died 1 times
isyp has died 3 times
POGXX has died 1 times
H3LLPER has died 1 times
yandere has died 6 times
Benpowell has died 5 times
Jalkas has died 1 times
KarlRasur has died 3 times
Mup has died 1 times
NeonTheDane has died 3 times
elhareen has died 1 times
Adoplaydoh has died 3 times
leoeo has died 1 times
Vast7al has died 2 times
Kekz has died 1 times
Frizuhrensohn has died 2 times
Bekokaan has died 11 times
Marraniello has died 3 times
Murcielago has died 3 times
General has died 2 times
Spoenky has died 7 times
Monark has died 2 times
Oona has died 2 times
Extreme0 has died 3 times
SPEEDY has died 3 times
Warp has died 6 times
Stoel has died 4 times
IMBA_seal has died 2 times
Kiosk has died 2 times
Doko has died 6 times
Igor-Ogir has died 1 times
Lanky has died 4 times
Sanchez has died 6 times
jokefox has died 3 times
Murphy has died 1 times
Baldurfreak has died 2 times
Coojee18 has died 1 times
Pole has died 1 times
Crystal has died 1 times
EnricoRics has died 2 times
inukami has died 1 times
Droopy has died 2 times
Spoof has died 2 times
Stenis has died 1 times
Jordan has died 1 times
Steve1596 has died 1 times
Savi has died 3 times
Grantel88 has died 4 times
iah has died 1 times
Zuck has died 2 times
Fumegatoru has died 1 times
Doom has died 1 times
Freaky314 has died 1 times
Captain has died 1 times
Cassis has died 1 times
Fabrixx1 has died 1 times
DrageBABA has died 1 times
Farpex has died 1 times
COCKBIG has died 1 times
Light has died 1 times
HazyClover has died 1 times
Itsame has died 1 times
lil_cey84 has died 2 times
Obi has died 1 times
midnight24 has died 1 times
Mikusis has died 2 times
Yung has died 2 times
Yaccer has died 1 times
grogo has died 1 times
Fosski has died 1 times
Ray has died 2 times
eliteguest45 has died 1 times
Rufiooh has died 1 times
boosted has died 4 times
Tiwaz has died 1 times
Yollo has died 1 times
spike has died 1 times
AriesNine has died 1 times
Mejo123 has died 3 times
Schrottspiel has died 2 times
trondl has died 1 times
asdfasdf has died 5 times
SHDY has died 1 times
Dsvrg has died 1 times
Braindead has died 1 times
nosegrinder has died 3 times
Rotary26b has died 1 times
Izaac314 has died 2 times
Efrador has died 2 times
Ezslay has died 1 times
Hank has died 1 times
aemkei has died 1 times
BlackStar has died 1 times
Aprelou has died 3 times
Drakkhan has died 1 times
Dixu has died 2 times
braz has died 1 times
Fordie has died 1 times
sermerlin has died 2 times
```
Is there any chance for this to be added to an embed, or better yet, a leaderboard that you can scroll on using buttons/reactions? And if possible the ability to sort the list by the highest deaths first.
Thank you!

**Comment by rfalias** [comment]
Oh damn, that is a big list!
Embed is limited to 6000 characters. 

Best bet is to probably chunk it and send as separate messages, would that work for you?

**Comment by AWL-Gaming** [comment]
> Oh damn, that is a big list! Embed is limited to 6000 characters.
> 
> Best bet is to probably chunk it and send as separate messages, would that work for you?

Yes that would work, it's better than no stats at all in my opinion.

**Comment by rfalias** [comment]
Fixed in #14 
Chunks the players to 100 player groups, they are also reverse sorted first so highest deaths are sent to discord first.
